src/Microsoft.AspNetCore.Blazor.Razor.Extensions/GenericComponentPass.cs

```diff
@@ -99,19 +219,86 @@ private void Process(ComponentExtensionNode node)
                         // the known types.
                         childContent.TypeName = RewriteTypeName(rewriter, childContent.TypeName);
                     }
+                    else if (childContent.IsParameterized)
+                    {
+                        // This is a weakly typed parameterized child content, treat it as RenderFragment<object>
+                        childContent.TypeName = BlazorApi.RenderFragment.FullTypeName + "<System.Object>";
+                    }
+                    else
+                    {
+                        // This is a weakly typed child content, treat it as RenderFragment
+                        childContent.TypeName = BlazorApi.RenderFragment.FullTypeName;
+                    }
                 }
             }
 
-            private string RewriteTypeName(GenericTypeNameRewriter rewriter, string typeName)
+            private string RewriteTypeName(CSharpSyntaxRewriter rewriter, string typeName)
             {
                 var parsed = SyntaxFactory.ParseTypeName(typeName);
                 var rewritten = (TypeSyntax)rewriter.Visit(parsed);
                 return rewritten.ToFullString();
             }
 
-            private string GetContent(ComponentTypeArgumentExtensionNode node)
+            private void CreateTypeInferenceMethod(
+                DocumentIntermediateNode documentNode,
+                ComponentExtensionNode node,
+                Dictionary<string, GenericTypeNameRewriter.Binding> bindings)
             {
-                return string.Join(string.Empty, node.FindDescendantNodes<IntermediateToken>().Where(t => t.IsCSharp).Select(t => t.Content));
+                var @namespace = documentNode.FindPrimaryNamespace().Content;
+                @namespace = string.IsNullOrEmpty(@namespace) ? "__Blazor" : "__Blazor." + @namespace;
+                @namespace += "." + documentNode.FindPrimaryClass().ClassName;
+
+                var typeInferenceNode = new ComponentTypeInferenceMethodIntermediateNode()
+                {
+                    Bindings = bindings,
+                    Component = node,
+
+                    // Method name is generated and guaraneteed not to collide, since its unique for each
```

Comment:
Typo: guaraneteed (and strictly speaking, its=>"it's" here, but I know that's kind of an annoying thing to mention)